File: 2022年8月/04 1403. 非递增顺序的最小子序列/minSubsequence.java
// 执行用时：3 ms, 在所有 Java 提交中击败了95.47%的用户
// 内存消耗：41.4 MB, 在所有 Java 提交中击败了81.30%的用户
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        Arrays.sort(nums);
        int length = nums.length;
        int sum = 0, resultSum = 0;
        List<Integer> list = new ArrayList<Integer>();
        for (int i : nums) {
            sum += i;
        }
        for (int i = length - 1; resultSum <= sum && i >= 0; i--) {
            list.add(nums[i]);
            resultSum += nums[i];
            sum -= nums[i];
        }
        return list;
    }
}

File: 2022年8月/04 1403. 非递增顺序的最小子序列/minSubsequence.js
/**
 * @param {number[]} nums
 * @return {number[]}
 * 执行用时：68 ms, 在所有 JavaScript 提交中击败了81.43%的用户
 * 内存消耗：43.3 MB, 在所有 JavaScript 提交中击败了77.14%的用户
 */
 var minSubsequence = function(nums) {
    nums.sort((a, b) => b - a);
    let sum = 0;
    let resultSum = 0;
    const length = nums.length;
    const result = [];
    for (let i = 0; i < length; i++) {
        sum += nums[i];
    }
    for (let i = 0; resultSum <= sum && i < length; i++) {
        result.push(nums[i]);
        resultSum += nums[i];
        sum -= nums[i];
    }
    return result;
};